_posts/2022-12-23-easily-verify-your-rust-in-ci-with-kani.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+title: "Easily verify your Rust in CI with Kani and Github Actions "
+---
+
+Formal verification tools, such as Kani, are valuable because they provide a way to definitively check, using mathematical techniques, whether a property of your code is true under all circumstances.  To put it colloquially: testing is great for catching the FIRST bug, formal verification is great for knowing you’ve caught the LAST bug.  
+
+Code, however, is seldom static. As code evolves, how can we ensure that it remains correct; i.e. how do we catch the NEXT bug? In the testing world, this is where **Continuous Integration** (CI) shines: by ensuring that tests are re-run on all code-changes, CI enables developers to catch bugs before they impact users.  Many platforms, such as GitHub, provide facilities to automatically incorporate CI into existing development workflows.  For example, [GitHub Actions](https://github.com/features/actions) provides developers with a facility to run checks, ranging from tests, to linters, to code-formatters, and have the results displayed on on the pull request (PR) in real time.  Repositories can even [require that certain checks pass before merging a PR](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging).  For example, on our own Kani repo, a CI status report on a PR might look like this:
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/kani-ci-checks.png" alt="Kani CI Checks" />
+
+Similarly, **Continuous Verification** enables developers to prevent regressions before they happen. Kani users have asked us for an easy way to integrate continuous verification into their existing development workflows. To this end, we are excited to announce the [Kani Rust Verifier Action](https://github.com/marketplace/actions/kani-rust-verifier) on the GitHub Marketplace.
+
+## How to use the Kani GitHub action
+
+If you have a [Rust Cargo](https://doc.rust-lang.org/cargo/) project in a GitHub repo, using the Kani CI action as simple as adding it to your `ci.yaml` file, where `<MAJOR>.<MINOR>` is the version of Kani you wish to use.  
```

Comment:
The `<MAJOR>.<MINOR>` part seems out of place?

Reply:
removed